larpandoracontent/LArThreeDReco/LArEventBuilding/TestBeamParticleCreationAlgorithm.cc

```diff
@@ -0,0 +1,94 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArEventBuilding/TestBeamParticleCreationAlgorithm.cc
+ * 
+ *  @brief  Implementation of the test beam particle creation algorithm class.
+ * 
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+
+#include "larpandoracontent/LArThreeDReco/LArEventBuilding/TestBeamParticleCreationAlgorithm.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+TestBeamParticleCreationAlgorithm::TestBeamParticleCreationAlgorithm() :
+    m_pfoListName("")
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode TestBeamParticleCreationAlgorithm::Run()
+{
+    const PfoList *pPfoList(nullptr);
+    PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::GetList(*this, m_pfoListName, pPfoList));
+
+    PfoList neutrinoPfos;
+
+    for (const Pfo *const pPfo : *pPfoList)
+    {
+        if (!LArPfoHelper::IsNeutrino(pPfo))
+            continue;
+
+        PfoList daughterList(pPfo->GetDaughterPfoList());
```

Comment:
Can make this const. If not using this list after deleting the neutrino Pfo, can make this a const reference.